core/lib/Thelia/Core/Template/Element/BaseLoop.php

```diff
@@ -396,13 +394,13 @@ protected function searchArray(array $search)
             return $search;
         }
 
-        $limit  = intval($this->getArgValue('limit'));
-        $offset = intval($this->getArgValue('offset'));
+        $limit  = (int) $this->getArgValue('limit');
+        $offset = (int) $this->getArgValue('offset');
 
         if ($this->getArgValue('page') !== null) {
-            $pageNum = intval($this->getArgValue('page'));
+            $pageNum = (int) $this->getArgValue('page');
 
-            $totalPageCount = ceil(count($search) / $limit);
+            $totalPageCount = ceil(\count($search) / $limit);
```

Comment:
Why don't you use \ceil() ?

Reply:
This function is not part of the list of functions optimized by opcode  

```
"array_slice"
"assert"
"boolval"
"call_user_func"
"call_user_func_array"
"chr"
"count"
"defined"
"doubleval"
"floatval"
"func_get_args"
"func_num_args"
"get_called_class"
"get_class"
"gettype"
"in_array"
"intval"
"is_array"
"is_bool"
"is_double"
"is_float"
"is_int"
"is_integer"
"is_long"
"is_null"
"is_object"
"is_real"
"is_resource"
"is_string"
"ord"
"strlen"
"strval"
```
https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3048
https://phpinternals.net/articles/optimising_internal_functions_via_new_opcode_instructions
https://www.youtube.com/watch?v=sliLs1hTvBc&index=5&t=9s&list=PLLG6JM2v8AKomfEOut67csdUTiTvqjYsM (25:00)